src/app/logic/player.js

```diff
@@ -0,0 +1,44 @@
+export class Player {
+  constructor() {
+    this.correctAnswersScore = 0;
+    this.amountOfAnsweredQuestions = 0;
+    this.detailQuestionData = [];
+    this.endGameData = {};
+  }
+
+  registerAnswer(correctAnswer, playerAnswer, isanswerCorrect) {
+    if (this.playerAnswer === false) {
+      throw new Error(
+        'error with playerAnswer, playerAnswer is null or undefined',
+      );
+    }
+    if (isanswerCorrect) {
+      this.correctAnswersScore += 1;
+    }
+    this.detailQuestionData.push({
+      playerAnswer,
+      isAnswerCorrect: isanswerCorrect,
+      answer: correctAnswer,
+    });
+    this.amountOfAnsweredQuestions += 1;
+  }
+
+  clearGameData() {
+    this.correctAnswersScore = 0;
+    this.amountOfAnsweredQuestions = 0;
+    this.detailQuestionData = [];
+  }
```

Comment:
So this seems like a bit of Player and Game (Quiz) information? Just give it a thought. Another option would be to create a new player everytime the game starts.
Also, have a look at the methods of clearing an array: https://stackoverflow.com/questions/1232040/how-do-i-empty-an-array-in-javascript